app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    collected_dictionary = {}
+
+    def wrapper(*args, **kwargs) -> dict:
+        dict_key = (func, args, frozenset(kwargs.items()))
+        if dict_key in collected_dictionary:
+            print("Getting from cache")
+            return collected_dictionary[dict_key]
+        else:
```

Comment:
else is redundant

Reply:
@pavlejviki could you please check my update "changed_solution_1"